minimint/src/consensus/interconnect.rs

```diff
@@ -37,11 +37,14 @@ fn call_internal<M: FederationModule + 'static>(
     method: Method,
     data: Value,
 ) -> http::Result<Response> {
-    let endpoint = module
+    let endpoint = match module
         .api_endpoints()
         .iter()
         .find(|endpoint| endpoint.method == method && endpoint.path_spec == path)
-        .ok_or_else(|| http::Error::from_str(404, "Endpoint not found"))?;
+    {
+        Some(e) => e,
+        None => return Ok(http::StatusCode::NOT_FOUND.into()),
+    };
```

Comment:
Could it be a problem that we return `Ok(http::StatusCode::NOT_FOUND.into())` instead of `Err(...)` ? basically this fn never returnes an Error now.